features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/admin_new/source/src/app/components/Throttling/Custom/AddEdit.jsx

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useReducer, useState } from 'react';
+import PropTypes from 'prop-types';
+import TextField from '@material-ui/core/TextField';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import { makeStyles } from '@material-ui/core/styles';
+import FormDialogBase from 'AppComponents/AdminPages/Addons/FormDialogBase';
+import { Typography } from '@material-ui/core';
+import API from 'AppData/api';
+import Alert from 'AppComponents/Shared/Alert';
+import MonacoEditor from 'react-monaco-editor';
+import sqlFormatter from 'sql-formatter';
+
+const useStyles = makeStyles((theme) => ({
+    error: {
+        color: theme.palette.error.dark,
+    },
+    dialog: {
+        minWidth: theme.spacing(150),
+
+    },
+    siddhiQueryHeading: {
+        marginTop: theme.spacing(2),
+        marginBottom: theme.spacing(1),
+    },
+    showSampleButton: {
+        marginTop: theme.spacing(2),
+    },
+}));
+
+const sampleSiddhiQuery = "FROM RequestStream SELECT userId, ( userId == 'admin@carbon.super' ) "
++ "AS isEligible , str:concat('admin@carbon.super','') as throttleKey "
++ 'INSERT INTO EligibilityStream; FROM EligibilityStream[isEligible==true]#throttler:timeBatch(1 min) '
```

Comment:
Since this is a hardcoded sample value, Can't we use a pre-formatted value hardcoded here, Otherwise adding a new dependency just for formatting this is not worth it, isn't it?

Reply:
I initially had a preformatted value here. But since we are using sql formatter to format the siddhi query received from the GET custom policy request I used that same formatter to format this text as well.